src/modding/ModHelpers.cs

```diff
@@ -0,0 +1,156 @@
+            case ModLoader.CheckErrorStatus.InvalidDependencyOrder:
+                result += string.Format(TranslationServer.Translate("MOD_ERROR_DEPENDENCIES_ORDER"), offendingMod.Name,
+                    string.IsNullOrWhiteSpace(otherMod.Name) ? otherMod.InternalName : otherMod.Name) + "\n";
+                result += string.Format(TranslationServer.Translate("MOD_ERROR_DEPENDENCIES_ORDER_FIX"),
+                    offendingMod.Name,
+                    string.IsNullOrWhiteSpace(otherMod.Name) ? otherMod.InternalName : otherMod.Name);
+                break;
+            case ModLoader.CheckErrorStatus.IncompatibleMod:
+                result += string.Format(TranslationServer.Translate("MOD_ERROR_INCOMPATIBLE_MOD"), offendingMod.Name,
+                    string.IsNullOrWhiteSpace(otherMod.Name) ? otherMod.InternalName : otherMod.Name) + "\n";
+                result += string.Format(TranslationServer.Translate("MOD_ERROR_INCOMPATIBLE_MOD_FIX"),
+                    string.IsNullOrWhiteSpace(otherMod.Name) ? otherMod.InternalName : otherMod.Name);
+                break;
+            case ModLoader.CheckErrorStatus.InvalidLoadOrderBefore:
+                result += string.Format(TranslationServer.Translate("MOD_ERROR_LOAD_ORDER_BEFORE"), offendingMod.Name,
+                    string.IsNullOrWhiteSpace(otherMod.Name) ? otherMod.InternalName : otherMod.Name) + "\n";
+                result += string.Format(TranslationServer.Translate("MOD_ERROR_LOAD_ORDER_BEFORE_FIX"),
+                    offendingMod.Name,
+                    string.IsNullOrWhiteSpace(otherMod.Name) ? otherMod.InternalName : otherMod.Name);
+                break;
+            case ModLoader.CheckErrorStatus.InvalidLoadOrderAfter:
+                result += string.Format(TranslationServer.Translate("MOD_ERROR_LOAD_ORDER_AFTER"), offendingMod.Name,
+                    otherMod.Name) + "\n";
+                result += string.Format(TranslationServer.Translate("MOD_ERROR_LOAD_ORDER_AFTER_FIX"),
+                    offendingMod.Name, otherMod.Name);
+                break;
+        }
+
+        return result;
+    }
+
+    public static VersionCompatibility GetVersionCompatibility(ModInfo info)
+    {
+        var isVersionAboveMin = false;
+        var isVersionBelowMax = false;
+
+        var isVersionMinDefined = !string.IsNullOrEmpty(info.MinimumThriveVersion);
+        var isVersionMaxDefined = !string.IsNullOrEmpty(info.MaximumThriveVersion);
+
+        if (isVersionMinDefined)
+        {
+            isVersionAboveMin = VersionUtils.Compare(Constants.Version, info.MinimumThriveVersion ?? string.Empty) >= 0;
+        }
+
+        if (isVersionMaxDefined)
+        {
+            isVersionBelowMax = VersionUtils.Compare(Constants.Version, info.MaximumThriveVersion ?? string.Empty) <= 0;
+        }
+
+        if ((isVersionAboveMin && isVersionMinDefined) || (isVersionBelowMax && isVersionMaxDefined))
+        {
+            return VersionCompatibility.Compatible;
```

Comment:
what happens when version is above min but also above max? for example:
`minVersion = "0.3.3"
maxVersion = "0.4.4"
version = "0.4.5"`
considering that minVersion is defined and 0.4.5 is above 0.3.3 first case is true which ignores second case due to the OR operator